dali/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_decoupled_api.h

```diff
@@ -93,6 +93,21 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
         nvjpegCreate(NVJPEG_BACKEND_HARDWARE, NULL, &handle_) == NVJPEG_STATUS_SUCCESS) {
       LOG_LINE << "Using NVJPEG_BACKEND_HARDWARE" << std::endl;
       NVJPEG_CALL(nvjpegJpegStateCreate(handle_, &state_hw_batched_));
+#if defined(NVJPEG_PREALLOCATE_API)
+      // TODO(awolant): How to expose chroma subsampling to the user?
+      if (spec.GetArgument<int>("preallocate_width_hint") > 0 &&
+          spec.GetArgument<int>("preallocate_height_hint") > 0) {
+        LOG_LINE << "Using NVJPEG_PREALLOCATE_API" << std::endl;
+        NVJPEG_CALL(nvjpegDecodeBatchedPreAllocate(
+          handle_,
+          state_hw_batched_,
+          hw_decoder_bs_,
```

Comment:
this might collide with the ongoing variable batch size changes. Maybe check with @szalpal

Reply:
Synced offline. Should be fine.